### PR TITLE
add createAccountWithSeed

### PR DESCRIPTION
**Pull request by rob-solana**

### CODE REVIEW

**Comment by rob-solana** [comment]
untested

where's the document that tells me how to run the tests locally?

File: src/layout.js [hunk 1]
```diff
@@ -43,5 +43,25 @@ export const rustString = (property: string = 'string') => {
     return _encode(data, buffer, offset);
   };
 
+  rsl.alloc = str => {
```
Comment (rob-solana) [inline src/layout.js]: here's the magic!

File: src/system-program.js [hunk 2]
```diff
@@ -139,13 +144,27 @@ const SystemInstructionLayout = Object.freeze({
...
+    ]),
+  },
 });
 
 /**
  * Populate a buffer of instruction data using the SystemInstructionType
  */
 function encodeData(type: SystemInstructionType, fields: Object): Buffer {
-  const data = Buffer.alloc(type.layout.span);
+  const allocLength =
```
Comment (rob-solana) [inline src/system-program.js]: more magic?

**Comment by rob-solana** [comment]
thank you @CriesofCarrots

File: src/system-program.js (hunk 2 above)
Comment (CriesofCarrots) [reply src/system-program.js]: Yes. I'm not entirely sure this is the best way to handle the the rust string, but it seemed reasonably extensible for other value-specific layouts, and I didn't see a more straightforward way using the `buffer-layout` package.

Comment (mvines) [reply src/system-program.js]: Can we just use a simple `char *` string?  Requiring utf8 is unnecessarily restrictive and is annoying to handle -- ie, all this code

Comment (CriesofCarrots) [reply src/system-program.js]: I'm not sure I see what you're envisioning, @mvines . The gymnastics in this pr are largely around handling dynamically sized layouts in buffer-layout, which would still be an issue even if not utf8, right?

Comment (mvines) [reply src/system-program.js]: I guess it seems wasteful to require a length of 8 bytes for a seed with a length of 32 bytes or less

Comment (CriesofCarrots) [reply src/system-program.js]: Ah, yeah. I expect we could use shortvec easily enough here, if @rob-solana likes that option upstream.

Comment (rob-solana) [reply src/system-program.js]: I've no preferences for how this side comes out...

for the system instruction, my goals are:
1. looks pretty for rust clients and for the implementation
2. supports a reasonable-sized mnemonic (32 is a nice length if one stays in ascii)
3. limit the space the seed can cover to something smaller than 256-bits (ascii and utf-8 have those restrictions)

**Pull request merged**